### DISCUSSION TITLE
A clarification on @skip / @include precedence

### DISCUSSION THREAD

**Discussion by BoD** [graphql/defer-stream-wg#31] [Q&A] [answered]
The spec edits mention `@include` and `@skip` take precedence over `@defer` and `@stream`.

I'm not 100% clear on what that means for a fragment with both `@defer(if: true) @include(if: true)` - should the defer directive be honored or not in this case?

**Comment by robrichard** [comment, marked as answer]
Yes the fragment should be deferred in that case. When`@defer(if: true)` is paired with `@include(if: false)` or `@skip(if: true)`, these fragment should be removed entirely and not deferred.

Do you have any suggestions on how the spec language can be clarified?

**Comment by BoD** [comment]
Thanks! That makes sense.

> how the spec language can be clarified?

Tough one! :)  Maybe something like "A fragment's execution should be delayed according to `@defer` even with`@include(if: true)` or `@skip(if: false)`"?

Or maybe the original sentence could be changed to "`@include(if: false)` and `@skip(if: true)` take precedence over `@defer`", if that makes any sense.
